featurebyte/api/target.py

```diff
@@ -0,0 +1,82 @@
+"""
+Target API object
+"""
+from __future__ import annotations
+
+from typing import Any, List, Optional
+
+from pydantic import Field
+
+from featurebyte.api.api_object import ForeignKeyMapping
+from featurebyte.api.entity import Entity
+from featurebyte.api.savable_api_object import SavableApiObject
+from featurebyte.models.base import PydanticObjectId
+from featurebyte.models.target import TargetModel
+from featurebyte.schema.target import TargetUpdate
+
+
+class Target(SavableApiObject):
+    """
+    Target class used to represent a Target in FeatureByte.
+    """
+
+    internal_entity_ids: Optional[List[PydanticObjectId]] = Field(alias="entity_ids")
+    horizon: Optional[str]
+    blind_spot: Optional[str]
```

Comment:
Maybe we should use `StrictStr` here. This prevent the case where user do something like `horizon=1` (if `str` is used, 1 is converted to '1'. If `StrictStr` is used, it will trigger error if non-string type is provided).

Reply:
good point, will update.

Reply:
Currently we mostly used at API object layer & public api payload check.